src/Form/CoyoteImgBatchForm.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\coyote_img_desc\Form;
+
+require_once( __DIR__ . '/../../vendor/autoload.php');
+
+use Coyote\CoyoteApiClientHelperFunctions;
+use Coyote\Model\OrganizationModel;
+use Coyote\Model\ProfileModel;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\coyote_img_desc\Constants;
+use Drupal\coyote_img_desc\Helper\CoyoteMembershipHelper;
+use Drupal\coyote_img_desc\Util;
+use Drupal\coyote_img_desc\DB;
+use Drupal\node\Entity\Node;
+use Coyote\ContentHelper;
+
+class CoyoteImgBatchForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return $this->t('coyote_img_batch_form');
+  }
+
+  private static function isDefined(?string $var): bool
+  {
+    return !is_null($var) && strlen($var) > 0;
+  }
+
+  private function getProfileRole(?ProfileModel $profile): string
+  {
+    if (is_null($profile)) {
+      return null;
+    }
+
+    $highestRole = CoyoteMembershipHelper::getHighestMembershipRole($profile->getMemberships());
+
+    if (is_null($highestRole)) {
+      return null;
+    }
+
+    return $this->t("Owner: {$profile->getName()} ({$highestRole})");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $config = $this->config('coyote_img_desc.settings');
+    $token = $config->get('api_token');
+    $endpoint = $config->get('api_endpoint');
+    $resourceGroup = $config->get('api_resource_group');
+    $organizationId = $config->get('api_organization');
+    if (self::isDefined($token) && self::isDefined($endpoint)) {
+      $profile = CoyoteApiClientHelperFunctions::getProfile($endpoint, $token);
+      $role = self::getProfileRole($profile);
+    }
+     
+    $disabled = false;
+    if (!self::isDefined($token) || !self::isDefined($endpoint) || !self::isDefined($organizationId) || !self::isDefined($resourceGroup) || is_null($profile) || is_null($role)) {
```

Comment:
Might result in more intuitive code to abstract a long chain of `or` into its own function such as `private function hasValidBatchProcessingConfig(): bool` or something like that.